### DISCUSSION TITLE
Only search in dictionaries of the corresponding language?

### DISCUSSION THREAD

**Discussion by Steven630** [koreader/koreader#12505] [Q&A]
It seems that when looking up a word, KOreader searches all the dictionaries, even those for different languages. Sometimes my Chinese word look-up also returns entries from Korean dictionaries. This slows down my kpw2 significantly. I have 7 dictionaries for three languages. Can I limit the search to dictionaries that has the same language so that when I look up an English word, it does not search in the 4 dictionaries whose source language is not English? This can be determined by the book metadata. If a Chinese book contains some citations from English sources, and an English look-up is needed (this is rare), the user can choose to "search in other dictionaries" to enable searching in dictionaries of other languages.

**Comment by dtcyganok** [comment]
That looks like a really good FR. I'd benefit of such a feature as well.

**Reply by Frenzie** [reply]
You can set dictionaries on a per-document basis.

**Reply by dtcyganok** [reply]
I suppose I can) Although having the checkbox "Limit dictionary search to current book language" would be more preferable.

**Reply by Frenzie** [reply]
Once in a blue moon a book has such metadata, a dictionary doesn't regardless (but perhaps one might be able to come up with a clever workaround with folders).

**Reply by Steven630** [reply]
All my books have language info in the metadata. Or allow the users to set the language once and for all?
